dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java

```diff
@@ -0,0 +1,354 @@
+        }
+    }
+
+    private BasicCredentialsProvider getCredentialsProvider() throws IOException, GeneralSecurityException {
+        final String esAuthType = getESAuthType();
+
+        //Loading TLS certificates
+        if (Config.getBooleanProperty("ES_TLS_ENABLED", true)) {
+            loadTLSCertificates();
+        }
+
+        final BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+
+        //Loading basic credentials if set
+        if (esAuthType.equals(BASIC_AUTH_TYPE)) {
+            credentialsProvider.setCredentials(AuthScope.ANY,
+                    new UsernamePasswordCredentials(
+                            Config.getStringProperty("ES_AUTH_BASIC_USER", null),
+                            Config.getStringProperty("ES_AUTH_BASIC_PASSWORD", null)));
+            Logger.info(DotRestHighLevelClientProvider.class,
+                    "Initializing Elastic RestHighLevelClient using Basic authentication");
+        }
+
+        return credentialsProvider;
+    }
+
+    private RestClientBuilder getClientBuilder(final BasicCredentialsProvider credentialsProvider) {
+        final String esAuthType = getESAuthType();
+
+        final RestClientBuilder clientBuilder = RestClient
+                .builder(new HttpHost(Config.getStringProperty("ES_HOSTNAME", "127.0.0.1"),
+                        Config.getIntProperty("ES_PORT", 19200), "https"))
+                .setHttpClientConfigCallback((httpClientBuilder) -> {
+                    if (sslContextFromPem != null) {
+                        httpClientBuilder
+                                .setSSLContext(sslContextFromPem);
+                    }
+                    httpClientBuilder
+                            .setDefaultCredentialsProvider(credentialsProvider);
+                    return httpClientBuilder;
+                })
+                .setFailureListener(new RestClient.FailureListener() {
+                    public void onFailure(Node node) {
+                        Logger.error(this, node.toString());
+                    }
+                });
+
+        //Loading token if JWT authentication is set
+        if (esAuthType.equals(JWT_AUTH_TYPE)) {
+            clientBuilder.setDefaultHeaders(new Header[]{new BasicHeader("Authorization",
+                    "Bearer " + Config.getStringProperty("ES_AUTH_JWT_TOKEN", null))});
+            Logger.info(DotRestHighLevelClientProvider.class,
+                    "Initializing Elastic RestHighLevelClient using JWT authentication");
+        }
+
+        return clientBuilder;
+    }
+
+    private String getESAuthType() {
+        return Config.getStringProperty("ES_AUTH_TYPE", BASIC_AUTH_TYPE);
+    }
+
+    private void loadTLSCertificates() throws IOException, GeneralSecurityException {
+        String clientCertPath = getCertPath("ES_AUTH_TLS_CLIENT_CERT", "elasticsearch.pem");
+        String clientKeyPath = getCertPath("ES_AUTH_TLS_CLIENT_KEY", "elasticsearch.key");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'clientKeyPath' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=3712629)